### PR TITLE
[Narwhal] remove NarwhalManager, LazyNarwhalClient, TransactionValidator and a few other integrations

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Remove a few integration points between Narwhal and Sui.
Remove some usages of Narwhal types in Sui.

## Test plan 

CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-tool/src/db_tool/mod.rs [hunk 1]
```diff
@@ -264,15 +263,11 @@ pub fn print_last_consensus_index(path: &Path) -> anyhow::Result<()> {
...
-pub fn print_consensus_commit(path: &Path, opt: PrintConsensusCommitOptions) -> anyhow::Result<()> {
-    let consensus_db = NodeStorage::reopen(path, None);
-    let consensus_commit = consensus_db
-        .consensus_store
-        .read_consensus_commit(&opt.seqnum)?;
-    match consensus_commit {
-        Some(commit) => println!("Consensus commit at {} is {:?}", opt.seqnum, commit),
-        None => println!("Consensus commit at {} is not found!", opt.seqnum),
-    }
+pub fn print_consensus_commit(
```
Comment (arun-koshy) [inline crates/sui-tool/src/db_tool/mod.rs:266]: Add a todo to remove this?

File: crates/sui-core/src/consensus_adapter.rs [hunk 2]
```diff
@@ -1133,8 +1097,8 @@ mod adapter_tests {
     use super::position_submit_certificate;
     use crate::consensus_adapter::{
         ConnectionMonitorStatusForTests, ConsensusAdapter, ConsensusAdapterMetrics,
-        LazyNarwhalClient,
     };
+    use crate::mysticeti_adapter::LazyMysticetiClient;
```
Comment (arun-koshy) [inline crates/sui-core/src/consensus_adapter.rs:1101]: thoughts on renaming to `LazyConsensusClient`?

File: crates/sui-core/src/consensus_manager/mod.rs [hunk 3]
```diff
@@ -201,26 +127,11 @@ impl ConsensusManagerTrait for ConsensusManager {
...
-                ConsensusProtocol::Mysticeti => {
-                    active[1] = true;
-                    self.consensus_client.set(self.mysticeti_client.clone());
-                    &self.mysticeti_manager
-                }
-            }
+            assert!(!*active, "Cannot start consensus. It is already running!");
+            info!("Starting consensus ...");
+            *active = true;
+            self.consensus_client.set(self.mysticeti_client.clone());
```
Comment (arun-koshy) [inline crates/sui-core/src/consensus_manager/mod.rs:133]: Do we need to do this on start? Can we just set consensus client when we create ConsensusManager and then we dont need mysticeti_client and consensus_client to be a part of the struct

**Review by arun-koshy — APPROVED** [review]
I think a nice followup would be to remove the ConsensusManagerTrait and ProtocolManager completely and just have ConsensusManager (MysticetiManager) module be used directly

File: crates/sui-core/src/consensus_adapter.rs (hunk 2 above)
Comment (mwtian) [reply crates/sui-core/src/consensus_adapter.rs:1101]: Added a TODO.

File: crates/sui-core/src/consensus_manager/mod.rs (hunk 3 above)
Comment (mwtian) [reply crates/sui-core/src/consensus_manager/mod.rs:133]: Before we decide on that we will never have a mode switching consensus between epoch, leaving the logic seems bettter.

File: crates/sui-tool/src/db_tool/mod.rs (hunk 1 above)
Comment (mwtian) [reply crates/sui-tool/src/db_tool/mod.rs:267]: Added a TODO to implement this later.

**Comment by mwtian** [comment]
> I think a nice followup would be to remove the ConsensusManagerTrait and ProtocolManager completely and just have ConsensusManager (MysticetiManager) module be used directly

We can definite do this once we are sure that we will not need to switch consensus implementations on epoch boundaries.

**Pull request merged**